deps/rabbit/src/rabbit_msg_store.erl

```diff
@@ -1220,56 +1548,64 @@ write_message(MsgId, Msg,
-                             {Msg1, State}
-                     end,
-                 gen_server2:reply(From, {ok, Msg}),
-                 State1;
-        false -> [#file_summary { locked = Locked }] =
-                     ets:lookup(FileSummaryEts, File),
-                 case Locked of
-                     true  -> add_to_pending_gc_completion({read, MsgId, From},
-                                                           File, State);
-                     false -> {Msg, State1} = read_from_disk(MsgLoc, State),
-                              gen_server2:reply(From, {ok, Msg}),
-                              State1
-                 end
-    end.
+%% @todo This shouldn't be possible anymore if reads are guaranteed to succeed on the client side. DONENOW
+%read_message(MsgId, From, State) ->
+%    case index_lookup_positive_ref_count(MsgId, State) of
+%        not_found   -> gen_server2:reply(From, not_found),
+%                       State;
+%        MsgLocation -> read_message1(From, MsgLocation, State)
+%    end.
+%
+%read_message1(From, #msg_location { msg_id = MsgId, file = File,
+%                                    offset = Offset } = MsgLoc,
+%              State = #msstate { current_file        = CurFile,
+%                                 current_file_handle = CurHdl,
+%                                 file_summary_ets    = FileSummaryEts,
+%                                 cur_file_cache_ets  = CurFileCacheEts }) ->
+%    case File =:= CurFile of
+%        true  -> {Msg, State1} =
+%                     %% can return [] if msg in file existed on startup
+%                     case ets:lookup(CurFileCacheEts, MsgId) of
+%                         [] ->
+%                             %% @todo This is no longer necessary if using the same handle as writing.
+%                             {ok, RawOffSet} =
+%                                 file_handle_cache:current_raw_offset(CurHdl),
+%                             ok = case Offset >= RawOffSet of
+%                                      true  -> file_handle_cache:flush(CurHdl);
+%                                      false -> ok
+%                                  end,
+%                             read_from_disk(MsgLoc, State);
+%                         [{MsgId, Msg1, _CacheRefCount}] ->
+%                             {Msg1, State}
+%                     end,
+%                 gen_server2:reply(From, {ok, Msg}),
+%                 State1;
+%        false -> [#file_summary { locked = Locked }] =
+%                     ets:lookup(FileSummaryEts, File),
+%                 case Locked of
+%                     true  -> add_to_pending_gc_completion({read, MsgId, From},
+%                                                           File, State);
+%                     false -> {Msg, State1} = read_from_disk(MsgLoc, State),
+%                              gen_server2:reply(From, {ok, Msg}),
+%                              State1
+%                 end
+%    end.
```

Comment:
might need to be brought back if we can call the store to read as is currently possible (though the clause is disabled)